perfdash/google-gcs-downloader.go

```diff
@@ -120,33 +114,57 @@ func (g *GoogleGCSDownloader) getJobData(wg *sync.WaitGroup, result JobToCategor
 		for categoryLabel, categoryMap := range tests.Descriptions {
 			for testLabel, testDescriptions := range categoryMap {
 				for _, testDescription := range testDescriptions {
-					fileStem := fmt.Sprintf("artifacts/%v_%v", testDescription.OutputFilePrefix, testDescription.Name)
-					artifacts, err := g.GoogleGCSBucketUtils.listFilesInBuild(job, buildNumber, fileStem)
+					filePrefix := fmt.Sprintf("%v_%v", testDescription.OutputFilePrefix, testDescription.Name)
+					searchPrefix := fmt.Sprintf("artifacts/%v", filePrefix)
+					artifacts, err := g.GoogleGCSBucketUtils.listFilesInBuild(job, buildNumber, searchPrefix)
 					if err != nil || len(artifacts) == 0 {
-						fmt.Printf("Error while looking for %s* in %s build %v: %v\n", fileStem, job, buildNumber, err)
+						fmt.Printf("Error while looking for %s* in %s build %v: %v\n", searchPrefix, job, buildNumber, err)
 						continue
 					}
-					metricsFilename := artifacts[0][strings.LastIndex(artifacts[0], "/")+1:]
-					if len(artifacts) > 1 {
-						fmt.Printf("WARNING: found multiple %s files with data, reading only one: %s\n", fileStem, metricsFilename)
+					for _, artifact := range artifacts {
+						metricsFileName := artifact[strings.LastIndex(artifact, "/")+1:]
+						resultCategory := getResultCategory(metricsFileName, filePrefix, categoryLabel, artifacts)
+						testDataResponse, err := g.GoogleGCSBucketUtils.getFileFromJenkinsGoogleBucket(job, buildNumber,
+							fmt.Sprintf("artifacts/%v", metricsFileName))
+						if err != nil {
+							fmt.Fprintf(os.Stderr, "Error when reading response Body: %v\n", err)
+							continue
+						}
+						buildData := getBuildData(result, tests.Prefix, resultCategory, testLabel, job, resultLock)
+						testDescription.Parser(testDataResponse, buildNumber, buildData)
 					}
-
-					testDataResponse, err := g.GoogleGCSBucketUtils.getFileFromJenkinsGoogleBucket(job, buildNumber, fmt.Sprintf("artifacts/%v", metricsFilename))
-					if err != nil {
-						fmt.Fprintf(os.Stderr, "Error when reading response Body: %v\n", err)
-						continue
-					}
-					resultLock.Lock()
-					buildData := result[tests.Prefix][categoryLabel][testLabel]
-					resultLock.Unlock()
-					testDescription.Parser(testDataResponse, buildNumber, buildData)
 					break
 				}
 			}
 		}
 	}
 }
 
+func getResultCategory(metricsFileName string, filePrefix string, category string, artifacts []string) string {
+	if len(artifacts) <= 1 {
+		return category
+	}
+	// if there are more artifacts, assume that this is a test suite run
+	trimmed := strings.TrimPrefix(metricsFileName, filePrefix+"_")
+	suiteId := strings.Split(trimmed, "_")[0]
```

Comment:
What if len(strings.Split(trimmed, "_")) == 0?

Reply:
from split documentation:
```
// If s does not contain sep and sep is not empty, Split returns a
// slice of length 1 whose only element is s.
(...)
func Split(s, sep string) []string { return genSplit(s, sep, 0, -1) }
```

Reply:
Good point.

Reply:
Can you give example of a file name and how will it behave when there is no suiteId?
file name  = "metric_test_timestamp" => suiteId = timestamp?

Reply:
Yes, that's correct. But this would mean that we'll have two files with same "metric_test" prefix and a different "timestamp" for the same run.

We could add some validation around it. For example we could expect that 
`len(strings.Split(trimmed, "_")) ` is greater than 2. However, I suggest to keep it simple for now. What do you think?

Reply:
> Yes, that's correct. But this would mean that we'll have two files with same "metric_test" prefix and a different "timestamp" for the same run.

What if there is only one file with this name format? Aren't all of our tests creating files like this unless we migrate them to test suite?
Example `ResourceUsageSummary_density_2019-10-02T10:08:21Z.json`

Reply:
For some reason I missed 
```
if len(artifacts) <= 1 {
	return category
}
```
It should behave correctly for old summaries.